### PR TITLE
[codex] Fix UniIR compatibility with transformers 5

### PR DESCRIPTION
**Pull request by lintool**
## Summary

- Add compatibility shims for UniIR tokenizer special-token accessors removed in `transformers` 5.
- Add `all_tied_weights_keys` compatibility for UniIR's BLIP MED `BertPreTrainedModel`.
- Apply the compatibility setup before UniIR query and corpus encoder construction.

## Validation

- `mamba run -n pyserini-dev3 python -m py_compile pyserini/encode/optional/_uniir.py`
- `mamba run -n pyserini-dev3 python -m unittest tests.optional.test_encoder_uniir`

The original search command now gets past UniIR encoder initialization and begins downloading the prebuilt indexes. Full CUDA validation was not possible locally because this env reports CUDA unavailable.

### CODE REVIEW

**Comment by lintool** [comment]
Follow-up debugging found two additional UniIR compatibility issues after the initial encoder construction fix:

- `transformers` 5 no longer exposes `get_head_mask`, which UniIR's MED `BertModel` still calls during forward passes.
- UniIR's single-query `QueryEncoder.encode()` wrapper is incompatible with its installed `encode_batch()` signature, so Pyserini now routes the single-query path through the batch API directly.

Additional validation:

- `mamba run -n pyserini-dev3 python -m py_compile pyserini/encode/optional/_uniir.py`
- `mamba run -n pyserini-dev3 python -m unittest tests.optional.test_encoder_uniir`
- A CPU debug run of the requested M-BEIR command with `--batch-size 2 --hits 1` processed 186 topics without Python traceback before I stopped it. This local environment has no CUDA, so the exact GPU path still was not exercised.

**Comment by lintool** [comment]
Added a follow-up fix for the `transformers` 5 `invert_attention_mask` deprecation warning. UniIR's MED model now gets a local compatibility override that preserves the existing mask conversion behavior without calling the deprecated Transformers helper.

Additional validation:

- `mamba run -n pyserini-dev3 python -m py_compile pyserini/encode/optional/_uniir.py`
- `mamba run -n pyserini-dev3 python -m unittest tests.optional.test_encoder_uniir`
- A short CPU warning-check run of the M-BEIR command with `--batch-size 2 --hits 1` processed 92 topics without emitting the `invert_attention_mask` warning before I stopped it.

**Review by clides — APPROVED** [review]
LGTM!

**Comment by lintool** [comment]
@sahel-sh @clides FYI - this completed fine on `orca`, with CPU:

```
Start time: 2026-06-06 15:36:50
End time: 2026-06-08 20:10:55
Total elapsed time: 189245s ~52.6hr
```
